### DISCUSSION TITLE
Unable to open .wf file

### DISCUSSION THREAD

**Discussion by gee4vee** [eclipse-glsp/glsp#599] [Q&A]
I set up a copy of the Node server and have it running:

```
~/g/D/blueprint-glsp-server-node ⛅» yarn start:server                                                              {14:08:11}
yarn run v1.22.17
$ yarn ts-node src/index.ts
$ /Users/gvalenc/git/DevX/blueprint-glsp-server-node/node_modules/.bin/ts-node src/index.ts
10:04:19 AM info -  [SocketServerLauncher] The GLSP server is ready to accept new client requests on port: 5007
[GLSP-Server]:Startup completed
```

I then start the `workflow-browser-app` from the client and it loads in the browser, but when I double click the `example1.wf` file nothing happens. This is the client output in the terminal (I'm assuming I can ignore the exception about starting the Java server since I'm using the Node one). Am I doing something wrong?

```
~/g/D/b/workflow-browser-app ⛅» yarn start:external                                                                {10:09:09}
yarn run v1.22.17
$ theia start --WF_GLSP=5007 --root-dir=../workspace --DEBUG
root ERROR Uncaught Exception:  Error: Could not launch GLSP server. The given jar path is not valid: /Users/gvalenc/git/DevX/glsp-server/target/org.eclipse.glsp.example.workflow-0.9.0-glsp.jar
root ERROR Error: Could not launch GLSP server. The given jar path is not valid: /Users/gvalenc/git/DevX/glsp-server/target/org.eclipse.glsp.example.workflow-0.9.0-glsp.jar
    at WorkflowGLServerContribution.launch (/Users/gvalenc/git/DevX/blueprint-glsp-client-node/node_modules/@eclipse-glsp/theia-integration/lib/node/java-socket-glsp-server.js:75:19)
    at GLSPBackendContribution.configure (/Users/gvalenc/git/DevX/blueprint-glsp-client-node/node_modules/@eclipse-glsp/theia-integration/lib/node/glsp-backend-contribution.js:51:30)
    at MessagingContribution.init (/Users/gvalenc/git/DevX/blueprint-glsp-client-node/node_modules/@theia/core/lib/node/messaging/messaging-contribution.js:55:26)
    at _postConstruct (/Users/gvalenc/git/DevX/blueprint-glsp-client-node/node_modules/inversify/lib/resolution/instantiation.js:33:31)
    at Object.resolveInstance (/Users/gvalenc/git/DevX/blueprint-glsp-client-node/node_modules/inversify/lib/resolution/instantiation.js:53:5)
    at /Users/gvalenc/git/DevX/blueprint-glsp-client-node/node_modules/inversify/lib/resolution/resolver.js:75:42
    at Object.resolve (/Users/gvalenc/git/DevX/blueprint-glsp-client-node/node_modules/inversify/lib/resolution/resolver.js:99:12)
    at /Users/gvalenc/git/DevX/blueprint-glsp-client-node/node_modules/inversify/lib/container/container.js:335:37
    at Container._get (/Users/gvalenc/git/DevX/blueprint-glsp-client-node/node_modules/inversify/lib/container/container.js:326:44)
    at Container.get (/Users/gvalenc/git/DevX/blueprint-glsp-client-node/node_modules/inversify/lib/container/container.js:243:21)
root INFO Theia app listening on http://localhost:3000.
root INFO Configuration directory URI: 'file:///Users/gvalenc/.theia'
root ERROR Cannot find a service for the path: /services/glsp/workflow/:id
root INFO Changed application state from 'init' to 'started_contributions'.
root INFO Changed application state from 'started_contributions' to 'attached_shell'.
root INFO >>> Restoring the layout state...
root INFO <<< The layout has been successfully restored.
root INFO Changed application state from 'attached_shell' to 'initialized_layout'.
root INFO Finished loading frontend application after 0.959 seconds
root INFO Changed application state from 'initialized_layout' to 'ready'.
root INFO Factory for model element 'comp:header' will be overwritten.
root INFO Factory for model element 'edge' will be overwritten.
```

**Comment by planger** [comment]
I'm sorry for the late reply!

It looks like the Theia backend doesn't respect the `--DEBUG` flag, according to the error you post. It tries to start a server instead of connecting to the one you started before (as it should once you pass the `--debug` argument). Briefly looking into the code, it seems that the [glsp-server-contribution](https://github.com/eclipse-glsp/glsp-theia-integration/blob/742564c31d2b5a55cd38ec124965598f6f148069/packages/theia-integration/src/node/glsp-server-contribution.ts#L96) actually checks for existence of the `--debug` argument via `string.startsWith('--debug')`, which is case sensitive. In most other cases we also set the argument in lower case (e.g. [in the example within the Theia integration code](https://github.com/eclipse-glsp/glsp-theia-integration/blob/185ac24a8248035bc602c7837569e8eec62b75d4/examples/browser-app/package.json#L26), etc.). However, in the [workflow-example of the glsp-examples repo](https://github.com/eclipse-glsp/glsp-examples/blob/e1c7994345d511e8799027d0ae10a6e1773aac0d/workflow/glsp-client/workflow-browser-app/package.json#L25), which is what you started, we use captials (`--DEBUG`).

Long story short, I believe this is a bug.
Can you please try if it works for you if you replace `--DEBUG` with `--debug` in the [workflow-example of the glsp-examples repo](https://github.com/eclipse-glsp/glsp-examples/blob/e1c7994345d511e8799027d0ae10a6e1773aac0d/workflow/glsp-client/workflow-browser-app/package.json#L25)?

Anyway, I've opened https://github.com/eclipse-glsp/glsp/issues/619 and propose to make this check case-insensitive.
